prow/cmd/sinker/main.go

```diff
@@ -109,9 +98,9 @@ func main() {
 	pushGateway := cfg().PushGateway
 	metrics.ExposeMetrics("sinker", pushGateway.Endpoint, pushGateway.Interval.Duration)
 
-	prowJobClient, err := o.kubernetes.ProwJobClient(cfg().ProwJobNamespace, o.dryRun.Value)
+	controlClusterCfg, err := o.kubernetes.ControlClusterConfig(o.dryRun.Value)
 	if err != nil {
-		logrus.WithError(err).Fatal("Error creating ProwJob client.")
+		logrus.WithError(err).Fatal("Error getting controlcluster config.")
 	}
 
 	buildClusterClients, err := o.kubernetes.BuildClusterClients(cfg().PodNamespace, o.dryRun.Value)
```

Comment:
After adding special handling and tests to ensure that sinker will only log messages and not do apicalls to Prowjobs when `dryRun` is enabled, I ralized that enabling that will make sinker crash here, as `BuildClusterClients` does not support it. Should I remove the code that handles `dryRun` again?

Reply:
I think we need to back away from the "DRY" `dryRun` behavior, it was applied too broadly and is not useful as you note.